domain/schema/controller.go

```diff
@@ -542,7 +542,7 @@ CREATE TABLE user (
     name            TEXT NOT NULL,
     display_name    TEXT,
     removed         BOOLEAN NOT NULL,
-    created_by_uuid TEXT,
+    created_by_uuid TEXT NOT NULL,
```

Comment:
I like this idea a lot. I have some thoughts on how we can possibly make this concept work really well for us.

During bootstrap I think we should introduce a `juju-bootstrap` user that is the nexus for all others. We will still make a `admin` user for the bootstrapping user.

In package `domain/user/bootstrap` I suspect we will have something like this:

```go
func  AddUser(user user.User) func(context.Context, db database.TxnRunner) error {
    // logic...
    return AddUserWithId(userId, user)
}

func AddUserWithId(user.Id, user.User) func(context.Context, db database.TxnRunner) error {
    // return logic func.
}

func CreateBootstrapUser() (user.Id, func(context.Context, db database.TxnRunner) error {
    // Create bootstrap user id
    return userId, AddUserWithId(userId, bootstrapUser)
}
```

The `juju-bootstrap` user now becomes the resource owner for anything created at bootstrap and owns the initial admin user that is setup as part of bootstrap.

Hope my ramblings make sense.